xdsl/traits.py

```diff
@@ -203,30 +203,51 @@ class SymbolOpInterface(OpTrait):
     a `SymbolTable` (TODO). A Symbol operation should use the SymbolOpInterface interface to
     provide the necessary verification and accessors.
 
-    Currently the only requirement is a "sym_name" attribute of type StringAttr.
+    A Symbol operation may be optional or not. If - the dault - it is not optional,
+    a `sym_name` attribute of type StringAttr is required. If it is optional,
+    the attribute is optional too.
 
-    Please see MLIR documentation for Symbol and SymbolTable for the requirements that are
-    upcoming in xDSL.
+    xDSL offers OptionalSymbolOpInterface as an always-optional SymbolOpInterface helper.
+
+    More requirements are defined in MLIR; Please see MLIR documentation for Symbol and
+    SymbolTable for the requirements that are upcoming in xDSL.
 
     https://mlir.llvm.org/docs/SymbolsAndSymbolTables/#symbol
     """
 
     @staticmethod
-    def get_sym_attr_name(op: Operation) -> StringAttr:
+    def get_sym_attr_name(op: Operation) -> StringAttr | None:
         """
-        Returns the symbol of the operation
+        Returns the symbol of the operation, if any
         """
         # import builtin here to avoid circular import
         from xdsl.dialects.builtin import StringAttr
 
+        concrete = op.get_trait(SymbolOpInterface)
+        assert concrete is not None
+        if concrete.is_optional_symbol(op) and "sym_name" not in op.attributes:
+            return None
         attr = op.attributes["sym_name"]
```

Comment:
Here, I would write first a check that `sym_name` is in `op.attributes`.
If it is not, then I would raise an exception (probably a `VerifyException`)

Reply:
This is the logic implemented in MLIR; what you say would raise a verifyexception on an optional symbol not being a symbol, or did I misunderstand ?

Reply:
Okay I think I was confused, will push something and see what you think :)

Reply:
Pushed something, WDYT?
Mh, thinking of it, maybe I should just call the verifier and return the thing or None? :thinking: